server/s3_put_object_action.cc

```diff
@@ -684,15 +703,30 @@ void S3PutObjectAction::add_part_object_to_probable_dead_oid_list(
             bucket_metadata->get_objects_version_list_index_layout().oid,
             object_metadata->get_version_key_in_index(),
             false /* force_delete */, part_entry[0].is_multipart, {0ULL, 0ULL},
-            1, key_indx,
+            1, it->first,
             bucket_metadata->get_extended_metadata_index_layout().oid,
             part_entry[0].versionID));
         parts_probable_del_rec_list.push_back(std::move(ext_del_rec));
+        it++;
       }
     }  // End of For
   }
 }
 
+void S3PutObjectAction::fetch_ext_object_info_success() {
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
+  add_object_oid_to_probable_dead_oid_list();
+  s3_log(S3_LOG_DEBUG, request_id, "%s Exit\n", __func__);
+}
+
+void S3PutObjectAction::fetch_ext_object_info_failed() {
+  // Add code in derived class to better handle
+  // failure in loading extended entries.
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
+  next();
```

Comment:
send Internal  error to client
Have logging

Reply:
Done